tests/system/feature_store/test_spark_engine.py

```diff
@@ -1422,6 +1422,164 @@ def test_ingest_with_steps_mapval(self, with_original_features):
+        measurements = fstore.FeatureSet(
+            "measurements_spark",
+            entities=[fstore.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "bad": {17: -1},
+                },
+                with_original_features=True,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        fstore.ingest(
+            measurements,
+            source,
+            targets,
+            spark_context=self.spark_service,
+            run_config=fstore.RunConfig(local=False),
+        )
+        csv_path_spark = measurements.get_target_path(name="csv")
+        df = self.read_csv(csv_path=csv_path_spark)
+        assert not df.empty
+        assert not df["bad_mapped"].isna().any()
+        assert not df["bad_mapped"].isnull().any()
+        assert not (df["bad_mapped"] == 17).any()
+        # There are no occurrences of -1 in the "bad" field of the original DataFrame.
+        assert (df["bad_mapped"] == -1).sum() == (original_df["bad"] == 17).sum()
+
+    def test_mapvalues_with_mixed_types(self):
+        key = "patient_id"
+        csv_path_spark = "v3io:///bigdata/test_mapvalues_with_mixed_types"
+        measurements = fstore.FeatureSet(
+            "measurements_spark",
+            entities=[fstore.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "hr_is_error": {True: "1"},
+                },
+                with_original_features=True,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        with pytest.raises(
+            mlrun.runtimes.utils.RunError,
+            match="^MapValues - mapping that changes column type must change all values accordingly,"
+            " which is not the case for column 'hr_is_error'$",
+        ):
+            fstore.ingest(
+                measurements,
+                source,
+                targets,
+                spark_context=self.spark_service,
+                run_config=fstore.RunConfig(local=False),
+            )
+
+    def test_mapvalues_validator(self):
+        key = "patient_id"
+        csv_path_spark = "v3io:///bigdata/test_mapvalues_validator"
+
+        measurements = fstore.FeatureSet(
+            "measurements_spark",
+            entities=[fstore.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "hr_is_error": {False: "no", True: 1},
+                },
```

Comment:
The type passed here does not match the parameter's type annotation of `mapping: Dict[str, Dict[str, Any]]`. One or the other should be fixed.

Reply:
fixed